collection-spec/json-schema/collection.json

```diff
@@ -74,14 +72,36 @@
               "type": "string"
             }
           }
+        },
+        "aggregation_control": {
+          "type": "object",
+          "required": ["variable_column_name"],
```

Comment:
Why is this required? I can imagine many scenarios where there is no variable name, such as dataset in which all the variables are output into a single netCDF.

What is special about "variable name"? Why can't it just be an anonymous attribute like everything else?

Reply:
When we merge/concat with xarray, we make everything that's not a variable into a coordinate to prevent xarray from inappropriately adding the time/ensemble dimension. I agree that this is something of kludge, but don't have another work around.

Reply:
This is avoidable with coords='minimal' and/or `data_vars='minimal'` Only variables with the specified concatenation dimension will be concatenated (the rest are merged, control equality comparison with `compat`). Though depending on the dataset, you may need expand_dims to make it work...

Reply:
I really don't think we want concepts like "variable" explicitly in the spec. Instead, we should address such merge conflicts via the `options`, as @dcherian suggested.

Basically, whatever you are doing internally in intake-esm to handle merging variables as a special case, bring that over to the collection JSON file in the `options` section.

Reply:
I agree with this in theory, but in practice we are finding that we need to be stashing variables as coordinates to accomplish the merge for now. At present we have failures like this
```
ValueError: {'time_bnds', 'lon_bnds', 'lat_bnds'} is a coordinate in some datasets but not others.
```

Reply:
this reflects anomalies in the datasets themselves

Reply:
Yeah this is because those variables are listed in the coordinates attribute in some files which is perfectly valid and correct.

I think `merge` is being unnecessarily strict. If it is not sure whether a variable is a coordinate or not, it should just leave that variable as a data variable.

We should fix this in xarray. Also `decode_coords=False` should work around it.

Reply:
For combining ensembles, we are actually using `concat` rather than `merge`. Is the error getting raised on the`concat` call or the`merge` call?

In any case, I think a good temporary workaround would be to add a special flag to `options` that triggers the operation you described:
> make everything that's not a variable into a coordinate to prevent xarray from inappropriately adding the time/ensemble dimension.